### PR TITLE
libc: Implement quick_exit and at_quick_exit 

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Required by new libcxx(https://github.com/apache/nuttx/pull/8173):
- libc: Flush all stream in _Exit 
- libc: abort should always call exit not pthread_exit
- libc: Move on_exit and __cxa_atexit in lib_atexit.c 
- libc: Implement quick_exit and at_quick_exit 

## Impact

New function
_Exit and abort

## Testing

Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@acassis @pkarashchenko could you review this patchset? #8173 is waiting for it.

File: libs/libc/stdlib/lib_exit.c [hunk 1]
```diff
@@ -46,22 +46,33 @@ FAR void *__dso_handle = &__dso_handle;
  * Public Functions
  ****************************************************************************/
 
-void exit(int status)
+void _Exit(int status)
```
Comment (pussuw) [inline libs/libc/stdlib/lib_exit.c:49]: I think flushing the streams here is not correct:

The _Exit() and _exit() functions shall not call functions registered with atexit() nor any registered signal handlers. Open streams shall not be flushed.  Whether open streams are closed (without flushing) is implementation-defined. Finally, the calling process shall be terminated with the consequences described below.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/_Exit.html

Comment (xiaoxiang781216) [reply libs/libc/stdlib/lib_exit.c]: You are right. the description in https://en.cppreference.com/w/c/program/_Exit isn't accuracy.

File: libs/libc/stdlib/lib_exit.c [hunk 2]
```diff
@@ -50,14 +50,25 @@ void exit(int status)
...
-  atexit_call_exitfuncs(status);
+  atexit_call_exitfuncs(status, false);
 
   /* Flush all streams */
 
   fflush(NULL);
 
   /* Then perform the exit */
 
+  _Exit(status);
```
Comment (pussuw) [inline libs/libc/stdlib/lib_exit.c:61]: Maybe change this to back to _exit() ?

Comment (xiaoxiang781216) [reply libs/libc/stdlib/lib_exit.c]: Done.

File: libs/libc/stdlib/lib_atexit.c [hunk 3]
```diff
@@ -181,3 +186,70 @@ int atexit(CODE void (*func)(void))
 {
   return atexit_register(ATTYPE_ATEXIT, func, NULL, NULL);
 }
+
+int at_quick_exit(CODE void (*func)(void))
```
Comment (pkarashchenko) [inline libs/libc/stdlib/lib_atexit.c:190]: Do we need to add a separate comment description for `at_quick_exit` or maybe modify existing description of `atexit`?

Comment (xiaoxiang781216) [reply libs/libc/stdlib/lib_atexit.c:205]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Review by pussuw — APPROVED** [review]
LGTM!

**Pull request merged**